core/client/src/client.rs

```diff
@@ -199,6 +201,13 @@ impl<B, E, Block> Client<B, E, Block> where
 			op.set_block_data(genesis_block.deconstruct().0, Some(vec![]), None, true)?;
 			backend.commit_operation(op)?;
 		}
+
+		// changes trie configuration should never change => we can read it in advance
+		let changes_trie_config = backend.state_at(BlockId::Number(Zero::zero()))?
```

Comment:
1. Yes at the moment. I haven't considered changing config as a high-priority feature
2. Should be possible - all important changes will be inside `state_machine/src/changes_trie`, except for additional requirement for light client to know about these changes when querying key changes proof (probably should be fetched with the proof itself)
3. Before last commit, there has been single check right before runtime method execution starts. I've added the same check at the end of execution (to make sure it can't be changed in block finalization code).